### ISSUE TITLE
Parser cannot handle 1. decimal syntax in SELECT

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#12824]
In various SQL dialects, `SELECT 1.` is valid decimal syntax, but jOOQ's parser doesn't recognise it:

```
Caused by: org.jooq.impl.ParserException: Token '*' expected: [1:10] SELECT 1.[*];
	at org.jooq.impl.DefaultParseContext.expected(ParserImpl.java:14018)
	at org.jooq.impl.DefaultParseContext.parse(ParserImpl.java:13278)
	at org.jooq.impl.DefaultParseContext.parse(ParserImpl.java:13273)
	at org.jooq.impl.DefaultParseContext.parseQualifiedAsteriskIf(ParserImpl.java:11758)
	at org.jooq.impl.DefaultParseContext.parseSelectList(ParserImpl.java:7326)
	at org.jooq.impl.DefaultParseContext.parseQueryPrimary(ParserImpl.java:1691)
	at org.jooq.impl.DefaultParseContext.parseQueryTerm(ParserImpl.java:1595)
	at org.jooq.impl.DefaultParseContext.parseQueryExpressionBody(ParserImpl.java:1561)
	at org.jooq.impl.DefaultParseContext.parseSelect(ParserImpl.java:1330)
	at org.jooq.impl.DefaultParseContext.parseSelect(ParserImpl.java:1325)
	at org.jooq.impl.DefaultParseContext.parseQuery(ParserImpl.java:1170)
	at org.jooq.impl.DefaultParseContext.lambda$0(ParserImpl.java:874)
	at org.jooq.impl.DefaultParseContext.wrap(ParserImpl.java:14230)
	at org.jooq.impl.DefaultParseContext.parse(ParserImpl.java:854)
	at org.jooq.impl.ParserImpl.parse(ParserImpl.java:732)
	at org.jooq.impl.ParserImpl.parse(ParserImpl.java:727)
	at org.jooq.impl.ParseFilesTest.parse(ParseFilesTest.java:106)
	... 36 more
```

**Comment by lukaseder** [comment]
The problem is not the decimal literal parsing, which works. E.g. this parses just fine:

```sql
SELECT 1
WHERE 1. = 1.
```

But we seem to be expecting an asterisk after the `.` because it's in the `SELECT` clause.

**Comment by lukaseder** [comment]
The error is in `parseIdentifierIf()`, which parses `1` as an identifier, despite most dialects requiring identifiers to start with a non-digit

**Comment by lukaseder** [comment]
When changing this, let's be careful not to break named parameters, such as `$1`, `:1`, `$1`

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.17.0 and 3.16.2 (https://github.com/jOOQ/jOOQ/issues/12827)
